spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/SkipperStreamService.java

```diff
@@ -0,0 +1,259 @@
+				new SpringCloudDeployerApplicationManifestReader().read(releaseManifest);
+
+		Map<String, SpringCloudDeployerApplicationManifest> appManifestMap = new HashMap<>();
+
+		for (SpringCloudDeployerApplicationManifest am : appManifests) {
+			String name = am.getSpec().getApplicationProperties().get("spring.cloud.dataflow.stream.app.label");
+			appManifestMap.put(name, am);
+		}
+
+		StreamDefinition streamDefinition = this.streamDefinitionRepository.findOne(streamName);
+
+		LinkedList<StreamAppDefinition> updatedStreamAppDefinitions = new LinkedList<>();
+		for (StreamAppDefinition appDefinition : streamDefinition.getAppDefinitions()) {
+			StreamAppDefinition.Builder appDefinitionBuilder = StreamAppDefinition.Builder.from(appDefinition);
+			SpringCloudDeployerApplicationManifest applicationManifest = appManifestMap.get(appDefinition.getName());
+			// overrides app definition properties with those from the release manifest
+			appDefinitionBuilder.setProperties(applicationManifest.getSpec().getApplicationProperties());
+			updatedStreamAppDefinitions.addLast(appDefinitionBuilder.build(streamDefinition.getName()));
+
+			//Check and register any missing app version
+			updateAppVersionIfChanged(appDefinition, applicationManifest);
+		}
+
+		String dslText = new StreamDefinitionToDslConverter().toDsl(updatedStreamAppDefinitions);
+
+		StreamDefinition updatedStreamDefinition = new StreamDefinition(streamName, dslText);
+		logger.debug("Updated StreamDefinition: " + updatedStreamDefinition);
+
+		// TODO consider adding an explicit UPDATE method to the streamDefRepository
+		// Note: Not transactional and can lead to loosing the stream definition
+		this.streamDefinitionRepository.delete(updatedStreamDefinition);
+		this.streamDefinitionRepository.save(updatedStreamDefinition);
+	}
+
+	private void updateAppVersionIfChanged(StreamAppDefinition appDefinition,
+			SpringCloudDeployerApplicationManifest appManifest) {
+
+		String version = appManifest.getSpec().getVersion();
+		//String name = appManifest.getSpec().getApplicationProperties().get("spring.cloud.dataflow.stream.app.label");
+		String name = appDefinition.getRegisteredAppName();
+		ApplicationType type = ApplicationType.valueOf(
+				appManifest.getSpec().getApplicationProperties().get("spring.cloud.dataflow.stream.app.type"));
+		String resource = appManifest.getSpec().getApplicationProperties().get("spring.cloud.dataflow.stream.app.type");
+
+		if (!this.appRegistryService.appExist(name, type, version)) {
+			URI metadataUri = null; // TODO Skipper should provide the metadata URI as well
+			this.appRegistryService.save(name, type, version, URI.create(resource), metadataUri);
+		}
+	}
+
+	@Override
+	public void doRollbackStream(String streamName, int releaseVersion) {
+		this.skipperStreamDeployer.rollbackStream(streamName, releaseVersion);
+	}
+
+	public String convertPropertiesToSkipperYaml(String streamName, Map<String, String> updateProperties) {
```

Comment:
Can we move this into somewhere into skipper domain (as it involves reading values from Skipper defined spec) and use it here? Maybe having it in ManifestUtils of Skipper would be better.

Reply:
This is a good idea, but since this code is not created by this PR and furthermore this suggestion would require and depend on committing code in another project (e.g. skipper)
So i would suggest to do this in separate ticket.
Btw the ManifestUtils is not accessible form scdf and will need to be moved as well.

Reply:
> This is a good idea, but since this code is not created by this PR and furthermore this suggestion would require and depend on committing code in another project (e.g. skipper)

+1

> ManifestUtils is not accessible form scdf and will need to be moved as well.

Yes, similar to what we did for other Manifest reader.